api/graphql/schema.py

```diff
@@ -54,32 +60,53 @@
 from permissions.api_permissions.graphene_permissions import (
     KeywordPermission,
     PurposePermission,
+    ReservationPermission,
     ReservationUnitPermission,
     ResourcePermission,
     SpacePermission,
     UnitPermission,
 )
+from permissions.helpers import (
+    get_service_sectors_where_can_view_reservations,
+    get_units_where_can_view_reservations,
+)
 from reservation_units.models import Equipment, EquipmentCategory, ReservationUnit
-from reservations.forms import ReservationForm
 from resources.models import Resource
 from spaces.models import Space, Unit
 
 
-class ReservationMutation(AuthMutation, DjangoModelFormMutation):
-    reservation = graphene.Field(ReservationType)
+class AllowAuthenticatedFilter(AuthFilter):
+    permission_classes = (AllowAuthenticated,)
+
 
+class ReservationsFilter(AuthFilter, django_filters.FilterSet):
     permission_classes = (
-        (ReservationUnitPermission,)
+        (ReservationPermission,)
         if not settings.TMP_PERMISSIONS_DISABLED
         else (AllowAny,)
     )
 
-    class Meta:
-        form_class = ReservationForm
-
-
-class AllowAuthenticatedFilter(AuthFilter):
-    permission_classes = (AllowAuthenticated,)
+    @classmethod
+    def resolve_queryset(
+        cls, connection, iterable, info, args, filtering_args, filterset_class
+    ):
+        queryset = super().resolve_queryset(
+            connection, iterable, info, args, filtering_args, filterset_class
+        )
+        user = (
+            get_user_model().objects.get(username="admin")
+            if settings.TMP_PERMISSIONS_DISABLED
+            else info.context.user
+        )
+        return queryset.filter(
+            Q(reservation_unit__unit__in=get_units_where_can_view_reservations(user))
+            | Q(
+                reservation_unit__unit__service_sectors__in=get_service_sectors_where_can_view_reservations(
+                    user
+                )
+            )
+            | Q(user=user)
+        ).order_by("begin")
```

Comment:
What do you think about doing something like this instead of relying on a specific user called `admin` to exist:
```python
user = info.context.user
viewable_units = get_units_where_can_view_reservations(user)
viewable_service_sectors = get_service_sectors_where_can_view_reservations(user)
if settings.TMP_PERMISSIONS_DISABLED:
    viewable_units = Unit.objects.all()
    viewable_service_sectors = ServiceSector.objects.all()
return queryset.filter(
    Q(reservation_unit__unit__in=viewable_units)
    | Q(reservation_unit__unit__service_sectors__in=viewable_service_sectors)
    | Q(user=user)
).order_by("begin")
```
Maybe an alternative would be to get the the first user with `is_superuser=True`, but I think it would be better not to assume any admin user exists.

Reply:
I actually need to include the user to get this TMP_PERMISSIONS_DISABLED to work as it supposed for now.